modules/host/negotiatedownload.go

```diff
@@ -93,7 +35,7 @@ func (h *Host) managedDownloadIteration(conn net.Conn, so *storageObligation) er
 	// The renter will either accept or reject the host's settings.
 	err = modules.ReadNegotiationAcceptance(conn)
 	if err != nil {
-		return err
+		return extendErr("renter rejected host settings: ", ErrorCommunication(err.Error()))
```

Comment:
For the most part, I tried to be conscious of two things when writing errors:

1. the error being returned may already be typed, but only if being returned by an internal-method call. So, `modules.ReadNegotiationAcceptance` won't return a typed error b/c it's not in the host package
2. the calling function is going to add context, so you don't need to. For example, if RPCFormContract calls RPCSettings, and RPCSettings returns an error, you'll get context up the whole stack. RPCSettings will extend with 'could not write secret key', RPCFormContract will extend with 'error during RPCSettings', and network.go will extend with 'error during RPCFormContract', so the full error ends up being:

"connection error: error during RPCFormContract: error during RPCSettings: could not write secret key: EOF" or something of that sort. You basically get the whole call stack in the form of context.